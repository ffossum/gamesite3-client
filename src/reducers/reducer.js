/* @flow */
import type { Action } from '../actions/actionTypes';

export type State = {};
const initialState = {};
export default function reducer(state?: State = initialState, action: Action) {
  switch (action.type) {
    default:
      return state;
  }
}
