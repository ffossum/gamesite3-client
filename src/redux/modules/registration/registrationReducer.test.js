/* @flow */
/* eslint-env jest */
import reducer from './registrationReducer';
import {
  registrationRequest,
  registrationSuccess,
  registrationFailure,
} from './registrationActions';

describe('registration reducer', () => {
  const registration = {
    username: 'bob',
    email: 'bob@test.com',
    password: 'bobisthebest',
    repeatPassword: 'bobisthebest',
  };
  const initialState = reducer(undefined, { type: '@@INIT' });
  test('initial state', () => {
    expect(initialState).toMatchSnapshot();
  });

  test('is loading after request', () => {
    const action = registrationRequest(registration);
    const state = reducer(initialState, action);
    expect(state.loading).toBe(true);
  });

  test('is no longer loading after success', () => {
    let state = reducer(initialState, registrationRequest(registration));
    state = reducer(
      state,
      registrationSuccess({
        id: 'abc123',
        username: 'bob',
      })
    );
    expect(state.loading).toBe(false);
  });

  test('is no longer loading after failure', () => {
    let state = reducer(initialState, registrationRequest(registration));
    state = reducer(state, registrationFailure());
    expect(state.loading).toBe(false);
  });
});
