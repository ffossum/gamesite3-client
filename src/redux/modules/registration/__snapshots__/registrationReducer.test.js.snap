// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registration reducer initial state 1`] = `
Object {
  "loading": false,
}
`;
